pkg/reconciler/model.go

```diff
@@ -112,3 +126,54 @@ type MothershipReconcilerConfig struct {
 	CrdComponents []string
 	PreComponents []string
 }
+
+type Repo struct {
+	URL   string `json:"url"`
+	Token string `json:"-"`
+}
+
+func (r *Repo) ReadToken(clientSet core.CoreV1Interface, namespace string) error {
+	secretKey, err := MapSecretKey(r.URL)
+	if err != nil {
+		return err
+	}
+
+	secret, err := clientSet.
+		Secrets(namespace).
+		Get(context.Background(), secretKey, v1.GetOptions{})
+
```

Comment:
QQ - this secret would be configured by SREs? If yes -we have to verify with them how we deal with it during the deployment.

Alternatively providing it as part of the comonent-configuration won't be an option, right (means, the credentials would be part of the component-configuration and the component-reconciler would create the secret in K8s if it's missing)?

Just thinking how we can avoid a coupling of our software to data in the underneath cluster. Also testing will become more difficult for us.